src/nvim/tui/input.c

```diff
@@ -47,7 +47,13 @@ void term_input_init(TermInput *input, Loop *loop)
   termkey_set_canonflags(input->tk, curflags | TERMKEY_CANON_DELBS);
   // setup input handle
 #ifdef WIN32
-  uv_tty_init(&loop->uv, &input->tty_in, 0, 1);
+  // On Windows, always read from the console input buffer instead of stdin.
+  // This avoids problems when stdin was previously used for other input.
+  const HANDLE conin_handle = CreateFile("CONIN$", GENERIC_READ | GENERIC_WRITE,
+                            FILE_SHARE_READ, (LPSECURITY_ATTRIBUTES)NULL,
+                            OPEN_EXISTING, 0, (HANDLE)NULL);
+  const int conin_fd = _open_osfhandle(conin_handle, _O_RDONLY);
+  uv_tty_init(loop, &input->tty_in, conin_fd, true);
```

Comment:
I think it's a better idea to move this patch to here because it's going to be less redundant. The patch won't need to be duplicated in #6299 if it's moved to here.

Reply:
@b-r-o-c-k seems reasonable, but then it won't work for `--headless` (TUI is not started then). So some duplication or refactoring might still be needed.

Reply:
You're right, this doesn't work for headless mode and I'm not sure if there's a good way to avoid the duplication so I'll just change it back to the way it was.